cypress/integration/GlossaryPopUps/GlossaryPopUps.js

```diff
@@ -0,0 +1,57 @@
+// <reference types="Cypress" />
+import { Given, And, Then } from 'cypress-cucumber-preprocessor/steps';
+
+Cypress.on('uncaught:exception', (err, runnable) => {
+    // returning false here to Cypress from
+    // failing the test
+    return false;
+});
+
+
+When('user is clicking on definition for {string}', (termName) => {
+    cy.get('a.definition').first().should('be.visible');
+    cy.get('a.definition').contains(termName).click();
+    cy.wait(500);
+
+});
+
+Then('pop-up definition box is displayed for {string}', (termName) => {
+    cy.on('window:alert', (termName) => {
+        expect('div.modal__container .term').to.contains('termName');
+    })
+});
+
+And('definition box displays all {string} provided by glossary', (features) => {
+    const expText = features.split(',');
+    cy.on('window:alert', (expText) => {
+        expect('div.modal__container').to.contains('expText');
+        cy.get('div.modal__container').as('definition');
+
+        for (let i = 0; i <= expText.length; i++) {
+            switch (expText[i]) {
+                case 0:
```

Comment:
Also switch case should specify the feature (e.g. 'definition' or 'pronunciation' or 'audio' or 'img') not the numbers - because feature file might not provide those features in exact order, I can add pronunciation,definition,img[2],audio and then it will fail